pyoverkiz/client.py

```diff
@@ -758,6 +760,61 @@ async def execute_scheduled_scenario(self, oid: str, timestamp: int) -> str:
         response = await self.__post(f"exec/schedule/{oid}/{timestamp}")
         return cast(str, response["triggerId"])
 
+    @backoff.on_exception(
+        backoff.expo,
+        (NotAuthenticatedException, ServerDisconnectedError),
+        max_tries=2,
+        on_backoff=relogin,
+    )
+    async def get_setup_options(self) -> list[Option]:
+        """
+        This operation returns all subscribed options of a given setup.
+        Per-session rate-limit : 1 calls per 1d period for this particular operation (bulk-load)
+        Access scope : Full enduser API access (enduser/*)
+        """
+        response = await self.__get("setup/options")
+        options = [Option(**o) for o in humps.decamelize(response)]
+
+        return options
+
+    @backoff.on_exception(
+        backoff.expo,
+        (NotAuthenticatedException, ServerDisconnectedError),
+        max_tries=2,
+        on_backoff=relogin,
+    )
+    async def get_setup_option(self, option: str) -> Option | None:
+        """
+        This operation returns the selected subscribed option of a given setup.
+        For example developerMode-{gateway_id} to understand if developer mode is on.
+        """
+        response = await self.__get(f"setup/options/{option}")
+
+        if response:
+            return Option(**humps.decamelize(response))
+
+        return None
+
+    @backoff.on_exception(
+        backoff.expo,
+        (NotAuthenticatedException, ServerDisconnectedError),
+        max_tries=2,
+        on_backoff=relogin,
+    )
+    async def get_setup_option_parameter(
```

Comment:
Nice!

Reply:
Except for the fact that this throws a very generic and ugly exception on the Overkiz side, if the option does not exist...